src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp

```diff
@@ -373,15 +373,12 @@ namespace Opm {
     static inline double critical_oil_gas( const SlgofTable& sgofTable ) {
 
         const auto& col = sgofTable.getKrogColumn();
-        using reverse = std::reverse_iterator< decltype( col.begin() ) >;
-        auto rbegin = reverse( col.begin() + sgofTable.numRows() );
-        auto rend = reverse( col.begin() );
-        const auto critical = std::upper_bound( rbegin, rend, 0.0 );
-        const auto index = std::distance( col.begin(), critical.base() - 1 );
-
-        if( critical == rend ) return 0.0;
-
-        return 1 - sgofTable.getSlColumn()[ index + 1 ];
+        const auto critical = std::upper_bound( col.begin(), col.end(), 0.0 );
+        const auto index = std::distance( col.begin(), critical - 1);
```

Comment:
Since you're essentially reimplementing the function, I suppose you could move the `std::distance()` calculation below the `critical == col.end()` check.